lib/dry/types/hash/key.rb

```diff
@@ -0,0 +1,60 @@
+require 'dry/equalizer'
+
+module Dry
+  module Types
+    class Hash < Definition
+      class Key
+        include Type
+        include ::Dry::Equalizer(:name, :type, :options)
+        include Decorator
+        include Builder
+
+        attr_reader :name
+
+        def initialize(type, name, required: Undefined, **options)
```

Comment:
nitpicking, but putting the name first in the params list would feel more natural to me

Reply:
to me as well, but the super constructor requires it in this order and changing the order of arguments when calling `super` is even more unnatural :)

Reply:
Oh! I didn't realise this. Fair enough then :)